### ISSUE TITLE
Trouble connecting CIDER to the repl

### ISSUE THREAD

**Issue by favetelinguis** [luminus-framework/luminus#104]
Im using luminus with +cljs and it is my understanding that this gives me two repls, one from lein figwheel wich gives my ClojureScript and one from lein run for Clojure.

For Clojure the documentation is found here:
http://www.luminusweb.net/docs/repl.md

For CLJS it is found here:
On yogthos.net blog.

However Im not able to use this documentation and get it working with my CIDER Emacs configuration.
The steps i do is:
1. C-c M-c
2. select localhost
3. select one of two running .lein:45128 (how can i connect the names here to the correct repl, for example figwheel vs clj repl?)

The error im getting is:
nREPL: Direct connection failed

I have included the nrepl-cider plugin under dev plugins.

Is there any documentation on using luimus with cider repl that im missing or can the existing documentation be updated to help me get this working?

**Comment by yogthos** [comment]
Yeah, Figwheel changed a bit since that post unfortunately. [Here's](https://github.com/bhauman/lein-figwheel/wiki/Running-figwheel-with-Emacs-Inferior-Clojure-Interaction-Mode) the official wiki for getting the REPL working.

**Comment by favetelinguis** [comment]
Thanks for the reply, however I want to use CIDER for this not inf-clojure, I always use cider for my day to day Clojure stuff.

Correct me if im wrong, but there are two nREPL sessions avaliable when working with a +cljs project. The figwheel one is only for the ClojureScript part and then I also have an ordinary CLJ REPL. As a default I should be able to connect to the CLJ one with localhost:7000 and figwheen localhost:7002?

**Comment by yogthos** [comment]
Yeah, the repl on `7000` is the one started by the app when you do `lein run` and the one on `7002` is started by the figwheel plugin. The second one is the one you want to use for cljs.

I use cursive myself, so I don't actually know much about CIDER setup for figwheel unfortunately.

**Comment by manandearth** [comment]
It is quite straight forward to add `cider-nrepl`, I could add it to the documentation and ask for a PR if it is required. basically embedding the Cider nREPL  is well documented in the [cider doc](http://docs.cider.mx/en/latest/installation/#using-embedded-nrepl-server)

It envolves adding the dependency and a handler.

**Comment by stardiviner** [comment]
Any updates on this?

**Comment by ragone** [comment]
This resolved my issue. https://github.com/clojure-emacs/cider/issues/2547#issuecomment-453755532
